### PR TITLE
Reset hystrix circuit when going back online

### PR DESCRIPTION
**Pull request by cammellos**
Wallet doesn't handle well going offline, this PR improves the situation.

It's difficult to test as you fallback on Infura, you can check logs for proper usage (you should not see any `open circuit` error after coming back online, while you will get them when offline).

To test:

1) Fetch balances
2) Go offline, fetch a few times balances
3) Go online, fetch balances

In develop:
- You will see `open circuits` in the logs after step 3) Here:
- You will not see `open circuits`

Basically on develop from when you go offline, you will be hitting infura directly for a while.
This PR fixes that behavior so we hit our proxy instead.

https://github.com/status-im/status-go/compare/d394da0f...10a12762

### CODE REVIEW

**Review by J-Son89 — APPROVED** [review]

**Comment by churik** [comment]
Checked behaviour, the cashng and building routes are the same for me, no entries with `circuit`, so I suppose it is fixed!
Also checked on 4G.

Thank you @cammellos

**Review by shivekkhurana — APPROVED** [review]

**Comment by ilmotta** [comment]
FYI With this other PR merged https://github.com/status-im/status-mobile/pull/21067, in theory, when the device is offline we won't even hit status-go endpoints `wallet_getWalletToken`, `wallet_getOwnedCollectiblesAsync`, and `wallet_checkRecentHistoryForChainIDs`. One PR doesn't invalidate the other, but after going offline the client won't try to fetch anything from status-go, thus nothing will hit the proxy or the Web3 providers.

**Review by ilmotta — APPROVED** [review]

**Comment by cammellos** [comment]
> FYI With this other PR merged #21067, in theory, when the device is offline we won't even hit status-go endpoints `wallet_getWalletToken`, `wallet_getOwnedCollectiblesAsync`, and `wallet_checkRecentHistoryForChainIDs`. One PR doesn't invalidate the other, but after going offline the client won't try to fetch anything from status-go, thus nothing will hit the proxy or the Web3 providers.

@ilmotta it will still most likely hit the proxy, I think (though not sure, but fairly certain) that status-go has some watchers that will hit the APIs in the background (those should be stopped in some subsequent prs if we know we are offline), so there's still one last piece of the puzzle :) but it should not matter for 2.30, the combination of your pr and mine should be enough to make it ok

**Comment by ilmotta** [comment]
> > FYI With this other PR merged #21067, in theory, when the device is offline we won't even hit status-go endpoints `wallet_getWalletToken`, `wallet_getOwnedCollectiblesAsync`, and `wallet_checkRecentHistoryForChainIDs`. One PR doesn't invalidate the other, but after going offline the client won't try to fetch anything from status-go, thus nothing will hit the proxy or the Web3 providers.
> 
> @ilmotta it will still most likely hit the proxy, I think (though not sure, but fairly certain) that status-go has some watchers that will hit the APIs in the background (those should be stopped in some subsequent prs if we know we are offline), so there's still one last piece of the puzzle :) but it should not matter for 2.30, the combination of your pr and mine should be enough to make it ok

Pardon my ignorance :) Correct, the event `:network/on-state-change` will notify `status-go`.

**Comment by cammellos** [comment]
closing as this has been merged both in develop & release
